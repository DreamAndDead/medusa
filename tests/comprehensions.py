a = [i * j for i in range(5) for j in range(3) if i * j % 2 == 0 and i > 0 and j > 0]

for item in a:
    print(item)

b = {i: i ** 2 for i in range(5)}

for k, v in b.items():
    print(k, v)