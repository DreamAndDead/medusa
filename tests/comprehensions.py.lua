dofile('./pythonlua/luainit.lua')

local a = (function() local result = list {} for i in range(5) do for j in range(3) do if (((math.fmod((i * j), 2)) == 0) and (i > 0)) then result.append((i * j)) end end end return result end)()
for item in a do
    print(item)
    ::loop_label_9::
end
local b = (function() local result = dict {} for i in range(5) do result[i] = (math.pow(i, 2)) end return result end)()
for k, v in b.items() do
    print(k, v)
    ::loop_label_10::
end