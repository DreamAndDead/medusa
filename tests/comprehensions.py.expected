2
2
4
6
6
4
8
12
10
1	1
2	4
3	9
4	16
5	25
